### PR TITLE
[hold][WIP] add tags widget to preprint form

### PR DESCRIPTION
**Pull request by ml3ha**
Add ember-osf's tags widget into the preprint form. This is a front-end only feature.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-    "font-awesome": "~4.6.3",
-    "jquery": "^2.2.4",
-    "pretender": "~1.1.0",
-    "Faker": "~3.1.0",
-    "bootstrap-treeview": "^1.2.0",
-    "dropzone": "^4.3.0",
-    "bootstrap-sass": "^3.3.6",
-    "toastr": "^2.1.2"
-  }
+    "name": "preprint-service",
```
Comment (caneruguz) [inline bower.json]: @ml3ha was this formatting change necessary? It's cluttering the PR

File: package.json [hunk 2]
```diff
@@ -46,6 +46,7 @@
     "ember-cli-test-loader": "^1.1.0",
     "ember-cli-uglify": "^1.2.0",
     "ember-collapsible-panel": "1.0.0",
+    "ember-cp-validations": "2.9.3",
```
Comment (caneruguz) [inline package.json]: Probably not part of this PR.

File: bower.json (hunk 1 above)
Comment (ml3ha) [inline bower.json]: @caneruguz The large differences here are because of formatting issues. When I initially added the jquery.tagsinput dependency, it was indented differently from everything else, which puzzled me. So I auto-indented the rest of the lines which is why it looks different.

Comment (abought) [inline bower.json]: Most JS tools indent things by 2 characters by default, but our style guide calls for 4.

Usually, we leave bower and package.json alone, because the tools that auto-regenerate those files do unpredictable things with indentation and don't always obey our rules. In general we just try to match the indentation that was already there to avoid noisy PRs.

File: app/templates/components/preprint-form-basics.hbs [hunk 3]
```diff
@@ -9,7 +9,7 @@
     </label>
     <label>
         Tags:
-        {{input class="form-control" value=tags}}
+        {{tags-widget}}
```
Comment (abought) [inline app/templates/components/preprint-form-basics.hbs]: h/t to @caneruguz for noticing: as written, this tags widget won't be able to do anything.

See documentation here. Essentially, the component must be passed three things:
http://abought.github.io/ember-osf/classes/tags-widget.html

1. A list of tags to display
2. An action that implements adding a tag
3. An action that implements removing a tag

See sample usage in the ember-osf dummy app:
https://github.com/abought/ember-osf/blob/19aee1f5ec2d6067b8fb9d80966994888d883335/tests/dummy/app/templates/nodes/detail/files/provider/file.hbs#L53-L53

File: ember-cli-build.js [hunk 4]
```diff
@@ -77,6 +77,8 @@ module.exports = function(defaults) {
         development: 'bower_components/bootstrap-treeview/src/js/bootstrap-treeview.js',
         production: 'bower_components/bootstrap-treeview/dist/js/bootstrap-treeview.min.js'
     });
+    app.import('bower_components/jquery.tagsinput/src/jquery.tagsinput.js');
```
Comment (abought) [inline ember-cli-build.js]: Sadly, this is necessary for now. 

It's not automatic because we didn't want every data model consumer to bloat their apps with our bower dependencies. Thanks for adding this. :)

**Comment by abought** [comment]
Done with review. As discussed on flowdock, the big blocker will be making sure that the tags widget is initialized with the correct arguments passed in.

**Comment by caneruguz** [comment]
Closing to redo with conflict resolution and fully working version.
